etl.py

```diff
@@ -49,55 +56,115 @@ def chunks(full_list, chunk_size, start_chunk=1, end_chunk=None):
 
 
 class UploadJob(object):
-    def __init__(self, csv_file_path, x_column="longitude", y_column="latitude", srid=4326):
+    def __init__(self, csv_file_path, x_column="longitude",
+                 y_column="latitude", srid=4326, file_encoding=FILE_ENCODING,
+                 force_no_geometry=FORCE_NO_GEOMETRY):
         self.csv_file_path = csv_file_path
         self.x_column = x_column
         self.y_column = y_column
         self.srid = srid
+        self.file_encoding = file_encoding
+        self.force_no_geometry = force_no_geometry
 
     def run(self):
         raise NotImplemented
 
     def regenerate_overviews(self):
-        query = 'select CDB_CreateOverviews(\'{table}\'::regclass)'.format(table=CARTO_TABLE_NAME)
+        query = 'select CDB_CreateOverviews(\'{table}\'::regclass)'.\
+            format(table=CARTO_TABLE_NAME)
         job_result = bsql.create(query)
         return job_result['job_id']
 
     def check_job(self, job_id):
         return bsql.read(job_id)
 
+    def create_geom_query(self, record):
+        null_result = NULL_VALUE + ","
+        if self.force_no_geometry:
+            return null_result
+
+        longitude = self.get_longitude(record)
+        latitude = self.get_latitude(record)
+
+        if longitude is None or latitude is None:
+            return null_result
+
+        return "st_transform(st_setsrid(st_makepoint(" + \
+            "{longitude}, {latitude}), {srid}), 4326),".\
+            format(longitude=longitude, latitude=latitude, srid=self.srid)
+
+    def parse_column_value(self, record, column):
+        result = None
+        try:
+            value = record[column]
+            if value is None or not value.strip():
+                result = NULL_VALUE + ","
+            else:
+                float(record[column])
+        except ValueError:
+            result = "'{value}',".format(value=value)
+        else:
+            if result is None:
+                result = "{value},".format(value=value)
+        return result
+
+    def get_longitude(self, record):
+        longitude = self.get_coord(record, self.x_column)
+        if abs(longitude) > MAX_LON:
+            return None
+        return longitude
+
+    def get_latitude(self, record):
+        latitude = self.get_coord(record, self.y_column)
+        if abs(latitude) > MAX_LAT:
+            return None
+        return latitude
+
+    def get_coord(self, record, type):
+        coord = record[type] or DEFAULT_COORD
+        try:
+            coord = float(coord)
+        except ValueError:
+            coord = DEFAULT_COORD
```

Comment:
not really that important, but for the sake of consistency, I think it should read `coord = float(DEFAULT_COORD)`

Reply:
setting DEFAULT_COORD as None as per a previous comment